app/Http/Controllers/Admin/UsersController.php

```diff
@@ -25,16 +34,58 @@ public function __construct()
     public function index()
     {
         return view('home');
-
     }
 
     public function edit(User $user)
     {
         return view('admin/users/edit', ['user' => $user, 'roles' => Role::all()]);
     }
 
+    /**
+     * Delete a user from the database
+     * @param int $userId The ID of the user that should be deleted
+     * @return view
+     */
+    public function delete($userId) {
+
+        // Validate the user ID
+        if (!User::where('id', $userId)->exists() || $userId == \Auth::user()->id) {
+            $users = User::all();
+            return view('admin/users/overview')->with('users', $users)->with('failure', true);
+        }
+
+        // Delete the user from the database and return to the overview
+        $result = \DB::table('users')->where('id', $userId)->delete();
+        $users = User::all();
+        return view('admin/users/overview')->with('users', $users)->with((!$result ? 'failure' : 'success'), true);
+    }
+
     public function overview()
     {
         return view('admin/users/overview', ['users' => User::all()]);
     }
-}
+
+    /**
+     * Upload an avatar
+     * @param file $file The file that should be uploaded
+     * @return boolean
+     */
+    private function uploadAvatar($avatar) {
```

Comment:
I'm afraid I haven't finished the avatar uploading function yet. It's not called on any page, so won't cause any errors. It'll be finished tonight/tomorrow.